lib/rabbitmq/cli/ctl/commands/encode_command.ex

```diff
@@ -0,0 +1,135 @@
+## The contents of this file are subject to the Mozilla Public License
+## Version 1.1 (the "License"); you may not use this file except in
+## compliance with the License. You may obtain a copy of the License
+## at http://www.mozilla.org/MPL/
+##
+## Software distributed under the License is distributed on an "AS IS"
+## basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See
+## the License for the specific language governing rights and
+## limitations under the License.
+##
+## The Original Code is RabbitMQ.
+##
+## The Initial Developer of the Original Code is GoPivotal, Inc.
+## Copyright (c) 2007-2016 Pivotal Software, Inc.  All rights reserved.
+
+defmodule RabbitMQ.CLI.Ctl.Commands.EncodeCommand do
+  @behaviour RabbitMQ.CLI.CommandBehaviour
+
+  def merge_defaults(args, opts) do
+    {args, Map.merge(%{
+        list_ciphers: false,
+        list_hashes:  false,
+        cipher:       :rabbit_pbe.default_cipher(),
+        hash:         :rabbit_pbe.default_hash(),
+        iterations:   :rabbit_pbe.default_iterations(),
+        decode:       false
+        }, opts)
+    }
+  end
+  def validate(_, %{list_ciphers: true, list_hashes: true}) do
+      {:validation_failure,
+       {:bad_argument, "Please list ciphers or hashes, not both."}}
+  end
+  def validate(args, %{list_ciphers: true, list_hashes: false}) when length(args) > 0 do
+      {:validation_failure,
+       {:bad_argument, :too_many_args}}
+  end
+  def validate(args, %{list_ciphers: false, list_hashes: true}) when length(args) > 0 do
+      {:validation_failure,
+       {:bad_argument, :too_many_args}}
+  end
+  def validate(args, %{list_ciphers: false, list_hashes: false}) when length(args) !== 2 do
+      {:validation_failure,
+       {:bad_argument, "Please provide a value to encode/decode and a passphrase."}}
+  end
+  def validate(args, %{list_ciphers: false, list_hashes: false} = opts) when length(args) === 2 do
+    case {supports_cipher(opts.cipher), supports_hash(opts.hash), opts.iterations > 0} do
+      {false, _, _}      -> {:validation_failure, {:bad_argument, "The requested cipher is not supported."}}
+      {_, false, _}      -> {:validation_failure, {:bad_argument, "The requested hash is not supported"}}
+      {_, _, false}      -> {:validation_failure, {:bad_argument, "The requested number of iterations is incorrect"}}
+      {true, true, true} -> :ok
+    end
+  end
+  def validate(_, _), do: :ok
+  def switches() do
+      [
+        list_ciphers: :boolean,
+        list_hashes: :boolean,
+        cipher: :atom,
+        hash: :atom,
+        iterations: :integer,
+        decode: :boolean
+      ]
+    end
+  def aliases(), do: []
+
+  def scopes(), do: [:ctl]
```

Comment:
`scopes/0` callback is optional. Default scope is calculated from a module name. For `RabbitMQ.CLI.Ctl` namespace it will be `:ctl`.